src/legacy/server/saved_objects/serialization/index.ts

```diff
@@ -64,12 +48,12 @@ export interface SavedObjectReference {
  */
 interface SavedObjectDoc {
   attributes: object;
-  id: string;
+  id?: string; // NOTE: SavedObjectDoc is used for uncreated objects where `id` is optional
   type: string;
   namespace?: string;
   migrationVersion?: MigrationVersion;
   version?: string;
-  updated_at?: Date;
+  updated_at?: string;
```

Comment:
`SavedObjectsRepository` sets `updated_at` to a `string` when creating/updating documents.